### DISCUSSION TITLE
Can I "defederate" other instance that my local users follow?

### DISCUSSION THREAD

**Discussion by Tuanle16111997** [mastodon/mastodon#16934] [Q&A]
Recently, my cache storage grow so quickly and I think that the reason is the federated timeline that make cache storage grow that fast. When I check the Remote account. It has reached to more than 1000 users while there are only under 50 local users. 

I have tried to suspend remote users but we can only do it one by one which is time-consuming. So I wonder if there is a way to remove the federate with other instance that my local users follow?

**Comment by nightpool** [comment]
Just so I understand.... you have local users who follow other users, and
you want to make it so those users can't follow users anymore? That seems
like it goes against the purpose of using Mastodon. I would suggest you
look into adding a block storage provider, like S3 or Backblaze, to deal
with media storage.

**Comment by LeeteqXV** [comment]
On Tue, Nov 2, 2021, at 4:25 AM, nightpool wrote:
> Just so I understand.... you have local users who follow other users, and
> you want to make it so those users can't follow users anymore? 

I think the OP is asking for a way to block the rest of those instances;
- allowing the users to get updates from the remote users they already follow, 
but to block the rest.

-- 
Regards,
Teqleez

**Reply by Tuanle16111997** [reply]
> On Tue, Nov 2, 2021, at 4:25 AM, nightpool wrote: Just so I understand.... you have local users who follow other users, and you want to make it so those users can't follow users anymore?
> I think the OP is asking for a way to block the rest of those instances; - allowing the users to get updates from the remote users they already follow, but to block the rest.
> […](#)
> -- Regards, Teqleez

Yes, it is exactly what I want to do. One local user can follow a lot of people from different instances which make federated timeline contains all of toots from these remote users while other people might don't need to see those things. The problem here is that federated timeline cost a lot of space (as I assume). so I want to somehow "defederate" these instances. only let federation relay work.

**Comment by nightpool** [comment]
Unless you're using a relay, every remote account in the federated timeline
is either followed by a local user, boosted by someone followed by a local
user, or replied to by someone followed by a local user. Broadly, these are
the only possible ways for a post to show up in the federated timeline.

**Reply by Tuanle16111997** [reply]
> Unless you're using a relay, every remote account in the federated timeline is either followed by a local user, boosted by someone followed by a local user, or replied to by someone followed by a local user. Broadly, these are the only possible ways for a post to show up in the federated timeline.
> […](#)
> On Tue, Nov 2, 2021, 6:37 AM Leeteq.XV ***@***.***> wrote: On Tue, Nov 2, 2021, at 4:25 AM, nightpool wrote: > Just so I understand.... you have local users who follow other users, and > you want to make it so those users can't follow users anymore? I think the OP is asking for a way to block the rest of those instances; - allowing the users to get updates from the remote users they already follow, but to block the rest. -- Regards, Teqleez — You are receiving this because you commented. Reply to this email directly, view it on GitHub <[#16934 (comment)](https://github.com/mastodon/mastodon/discussions/16934#discussioncomment-1575513)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AABZCV4Y6WRXTTOMKBF2FATUJ7EPBANCNFSM5HFNY4LA> . Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

I understand that mechanism. However, is there anyway to stop federated timeline from collecting toot from remote account? Users can see toots from the users that they follow on their home timeline. But other users of my instance might not need to see these toots on federated timeline. 
What I want to do is to narrow down the federated timeline. I want federated timeline to contains only toots from the instance that I want to relay. Is it possible? While each instance only focuses on one topic, why don't we federate with only the instances related to our topic?

**Reply by nightpool** [reply]
> One local user can follow a lot of people from different instances which make federated timeline contains all of toots from these instances

This is not true. Mastodon's federated timeline does not work in this way. The only posts you see are 

 1. posts by remote users who are followed by people on your server
 2. posts replied to or reblogged by remote users who are followed by people on your server.
